Doc build failure not noticed by CI test system

With the recent changes from using a Makefile to CMake, if the doc build fails (with new errors/warnings being reported) the CI build is still marked as passing even though there are errors/warnings:

```
New errors/warnings found, please fix them:

/home/dbkinder/zephyr/tests/kernel/queue/src/test_queue_user.c:22: warning: missing title after \defgroup kernel_queue_tests
/home/dbkinder/zephyr/include/net/ethernet.h:477: warning: argument '0' of command @param is not found in the argument list of net_eth_promisc_mode(struct net_if *iface, bool enable)
/home/dbkinder/zephyr/include/net/ethernet.h:483: warning: The following parameters of net_eth_promisc_mode(struct net_if *iface, bool enable) are not documented:
  parameter 'enable'
/home/dbkinder/zephyr/include/net/ethernet.h:477: warning: argument '0' of command @param is not found in the argument list of net_eth_promisc_mode(struct net_if *iface, bool enable)
/home/dbkinder/zephyr/include/net/ethernet.h:483: warning: The following parameters of net_eth_promisc_mode(struct net_if *iface, bool enable) are not documented:
  parameter 'enable'
/home/dbkinder/zephyr/include/net/ethernet.h:477: warning: argument '0' of command @param is not found in the argument list of net_eth_promisc_mode(struct net_if *iface, bool enable)
/home/dbkinder/zephyr/include/net/ethernet.h:483: warning: The following parameters of net_eth_promisc_mode(struct net_if *iface, bool enable) are not documented:
  parameter 'enable'
/home/dbkinder/zephyr/include/net/ethernet.h:477: warning: argument '0' of command @param is not found in the argument list of net_eth_promisc_mode(struct net_if *iface, bool enable)
/home/dbkinder/zephyr/include/net/ethernet.h:483: warning: The following parameters of net_eth_promisc_mode(struct net_if *iface, bool enable) are not documented:
  parameter 'enable'
```

--- a/.shippable.yml
+++ b/.shippable.yml
@@ -47,9 +47,9 @@ build:
               echo "Documentation build failed";
               exit 1;
             fi
-            if [ -s doc/doc.warnings ]; then
+            if [ -s doc/_build/doc.warnings ]; then
               echo " => New documentation warnings/errors";
-              cp doc/doc.warnings doc.warnings
+              cp doc/_build/doc.warnings doc.warnings
             fi;
             echo "- Verify commit message, coding style, doc build";
             ./scripts/ci/check-compliance.py --commits ${COMMIT_RANGE} || true;

